lib/lutaml/model/xml_adapter/nokogiri_adapter.rb

```diff
@@ -130,6 +134,10 @@ def build_ordered_element(xml, element, options = {})
         end
 
         def add_to_xml(xml, value, attribute, rule)
+          if rule.custom_methods[:to]
```

Comment:
I noticed in the MappingRule class there are serialize and deserialize methods that used the custom_methods. How about we consolidate value processing (custom_method, defaults) in the Attribute class?